contracts/modules/Experimental/Checkpoints/PLCRVotingCheckpoint.sol

```diff
@@ -0,0 +1,321 @@
+            now < uint256(ballots[_ballotId].startTime)
+            .add(uint256(ballots[_ballotId].commitDuration)
+            .add(uint256(ballots[_ballotId].revealDuration))),
+            "Already ended"
+        );
+        uint8 activeStatus = 0;
+        if (_isActive)
+            activeStatus = 1;
+        require(ballots[_ballotId].isActive != activeStatus, "Active state unchanged");
+        ballots[_ballotId].isActive = activeStatus;
+        emit BallotStatusChanged(_ballotId, _isActive);
+    }
+
+    /**
+     * @notice Used to get the current stage of the ballot
+     * @param _ballotId Given ballot Id
+     */
+    function getCurrentBallotStage(uint256 _ballotId) public view returns (Stage) {
+        Ballot memory ballot = ballots[_ballotId];
+        uint256 commitTimeEnd = uint256(ballot.startTime).add(uint256(ballot.commitDuration));
+        uint256 revealTimeEnd = commitTimeEnd.add(uint256(ballot.revealDuration));
+
+        if (now < ballot.startTime)
+            return Stage.PREP;
+        else if (now <= commitTimeEnd && now >= ballot.startTime)
+            return Stage.COMMIT;
+        else if ( now > commitTimeEnd && now <= revealTimeEnd)
+            return Stage.REVEAL;
+        else if (now > revealTimeEnd)
+            return Stage.RESOLVED;
+    }
+
+    /**
+     * @notice Queries the result of a given ballot
+     * @param _ballotId Id of the target ballot
+     * @return uint256 voteWeighting
+     * @return uint256 tieWith
+     * @return uint256 winningProposal
+     * @return bool isVotingSucceed
+     * @return uint256 totalVotes
+     */
+    function getBallotResults(uint256 _ballotId) external view returns(uint256[] memory, uint256[] memory, uint256, bool, uint256) {
+        if (_ballotId >= ballots.length)
+            return (new uint256[](0), new uint256[](0), 0, false, 0);
+        
+        Ballot storage ballot = ballots[_ballotId];
+        uint256 i = 0;
+        bool isVotingSucceed = false;
+        uint256 counter = 0;
+        uint256 maxWeight = 0;
+        uint256 winningProposal = 0;
+        uint256 quorumWeight = (ballot.totalSupply.mul(ballot.quorum)).div(10 ** 18);
+        uint256[] memory voteWeighting = new uint256[](ballot.totalProposals);
+        for (i = 0; i < ballot.totalProposals; i++) {
+            voteWeighting[i] = ballot.weightedVote[i];
+            if (maxWeight < ballot.weightedVote[i]) {
+                maxWeight = ballot.weightedVote[i];
+                if (maxWeight >= quorumWeight)
+                    winningProposal = i;
+            }
+        }
+        if (maxWeight >= quorumWeight) {
+            isVotingSucceed = true;
+            for (i = 0; i < ballot.totalProposals; i++) {
+                if (maxWeight == ballot.weightedVote[i] && i != winningProposal)
+                    counter ++;
+            }
+        }
+        
+        uint256[] memory tieWith = new uint256[](counter);
+        counter = 0;
+        if (counter > 1) {
```

Comment:
This will never be true because the counter is being set to 0 in the above line.
the counter should be set to 0 in the next line and not before. 
Also, the statement should be `counter >= 1` instead of `counter > 1`